cypher-shell/src/main/java/org/neo4j/shell/cli/ErrorPassingPrintStream.java

```diff
@@ -0,0 +1,168 @@
+package org.neo4j.shell.cli;
+
+import org.neo4j.shell.exception.JLineException;
+
+import java.io.IOException;
+import java.io.PrintStream;
+import java.util.Locale;
+
+/**
+ * This PrintStream will throw an exception when invoked. This is so that JLine gives us the error, instead of
+ * printing directly to STDERR.
+ */
+class ErrorPassingPrintStream extends PrintStream {
```

Comment:
This class is pretty hard to understand. Perhaps it would be clearer if the "other" methods didn't throw the exception too: I think you're suggesting that they will never be called, is that correct?